iotc-generic-c-sdk/include/iotconnect.h

```diff
@@ -16,6 +16,40 @@
 extern "C" {
 #endif
 
+#include <stdio.h>
+
+/*
+ * To use iotconnect need to specify debug routines IOTC_DEBUG, IOTC_WARN and IOTC_ERROR
+ *
+ * Two example implementations are shown below.
+ * Another possible implementation would be to use syslog().
+ */
+// #define USE_SYSLOG 1
+#if USE_SYSLOG
+#include <syslog.h>
+#define IOTC_ERROR(...) syslog(LOG_ERR, __VA_ARGS__)
+#define IOTC_WARN(...) syslog(LOG_WARNING, __VA_ARGS__)
+#define IOTC_DEBUG(...) syslog(LOG_DEBUG, __VA_ARGS__)
+#else
+#define IOTC_ENDLN "\n"
+
+#define IOTC_DEBUG_LEVEL 2
```

Comment:
Since we don't have the user config file header support, statements like these should allow a user overridel levels without without having to modify this file by passing a compile time flag like -DIOTC_DEBUG_LEVEL=1 . For example,
```c
#ifndef IOTC_DEBUG_LEVEL
#define IOTC_DEBUG_LEVEL 2
#endif
```